cleverpush/src/main/java/com/cleverpush/CleverPush.java

```diff
@@ -1515,14 +1522,34 @@ public void getAvailableTopics(ChannelTopicsListener listener) {
         return topics;
     }
 
+    public void addSubscriptionTags(String[] tagIds) {
+        this.tagIdsToAdd = tagIds;
+        addingMultipleTag = true;
+        currentPositionOfTagToAdd = 0;
+        addSubscriptionTag(tagIds[currentPositionOfTagToAdd]);
+    }
+
     public void addSubscriptionTag(String tagId) {
-        Log.d("CleverPush", "addSubscriptionTag: " + tagId);
         this.waitForTrackingConsent(() -> new Thread(() -> this.getSubscriptionId(subscriptionId -> {
             if (subscriptionId != null) {
                 Set<String> tags = this.getSubscriptionTags();
                 if (tags.contains(tagId)) {
-                    Log.d("CleverPush", "Subscription already has tag - skipping API call " + tagId);
-                    return;
+                    if (addingMultipleTag) {
```

Comment:
The logic for the bulk adding of tags should not happen in `addSubscriptionTag` - it should happen in the new `addSubscriptionTags` method.
Maybe you can add a callback inside here (`addSubscriptionTag(String tagId, CallbackListener callback)...` to check when the method is done so we will add one tag after each other.

Example:
1. Add Tag 1
2. Wait for callback
3. Add Tag 2.
4. Wait for callback
5. ...